File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> myMap;
        for(int i = 0; i < nums.size(); i++) {
            int f = target-nums[i];
            if(myMap.find(f) != myMap.end()) {
                return {myMap[f], i};
            }
            myMap[nums[i]] = i;
        }
        return {};
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> myMap = new HashMap<>();
        int []ans = new int[2];
        int n = nums.length;
        for(int i = 0; i < n; i++) {
            if(myMap.containsKey(target-nums[i])){
                ans[0] = myMap.get(target-nums[i]);
                ans[1] = i;
                return ans;
            }
            myMap.put(nums[i], i);
        }
        return ans;
    }
}